addons/website_event_track/static/src/js/website_event_track_proposal.js

```diff
@@ -0,0 +1,185 @@
+     * 
+     * @private
+     */
+    _manageRequiredContactFields: function() {
+        var $contactName = this.$(".o_wetrack_contact_name_input")[0];
+        if (this.useAdvancedContact) {
+            $contactName.setAttribute("required", "True");
+        } else {
+            $contactName.removeAttribute("required");
+        }
+    },
+
+    /**
+     * If there are still errors in form, display the error section and
+     * compose the error message accordingly.
+     * 
+     * @private
+     * @param {Array} errors - Names of errors still present in form.
+     */
+    _updateErrorDisplay: function(errors) {
+
+        if (!errors.length) {
+            this.$('.o_wetrack_proposal_error_section').addClass('d-none');
+        } else { 
+            this.$('.o_wetrack_proposal_error_section').removeClass('d-none');
+        }
+
+        var errorMessage = '';
+        var $errorElement = this.$('.o_wetrack_proposal_error_message');
+ 
+        if (errors.includes('invalidFormInputs')) errorMessage += ' Please fill out the form correctly.';
+        if (errors.includes('noContactMean')) errorMessage += ' Please enter a contact email address or a contact phone number.';
+ 
+        $errorElement.text(_t(errorMessage)).change();
+     },
+
+    /**
+     * Check validity of form input fields: 
+     * - 1) error 'invalidFormInputs' : Invalid ones are marked as is-invalid and o_has_error.
+     * - 2) error 'noContactMean' : Contact mean fields are marked as is-invalid and contact
+     * section as o_has_contact_mean_error if none of them is filled.
+     * Update isFormValid ( true if no error remain )
+     * 
+     * @private
+     */
+    _validateForm: function () {
+        var formErrors = [];
+        this._manageRequiredContactFields();
+
+        // 1) Valid Form Inputs
+        this.$('.form-group').each(function (index, field) { 
+            var $field = $(field);
+            // Validate current input, if not select2 field.
+            var inputs = $field.find('.form-control').not('.o_wetrack_select2_tags');
+            var invalid_inputs = inputs.toArray().filter(function (input) {
+                return !input.checkValidity();
+            });
+            
+            $field.find('.form-control').removeClass('o_has_error').removeClass('is-invalid');
+            if (invalid_inputs.length) {
+                $field.find('.form-control').addClass('o_has_error').addClass('is-invalid');
+                formErrors.push('invalidFormInputs');
+            }
+        });
+
+        // 2) Advanced Contact Must Have a Contact Mean
+        if (this.useAdvancedContact) {
+            var hasContactMean = this.$('.o_wetrack_contact_phone_input').val() || 
+                                 this.$('.o_wetrack_contact_email_input').val();
+            if (!hasContactMean) { 
+                this.$('.o_wetrack_contact_information').addClass('o_has_contact_mean_error'); 
```

Comment:
Classes are often pre-fixed with the module.
Like "o_wetrack_no_contact_prossibility_error".